lib/lita/adapters/slack/rtm_connection.rb

```diff
@@ -31,8 +31,15 @@ def initialize(robot, config, team_data)
           @websocket_url = team_data.websocket_url
           @robot_id = team_data.self.id
 
-          @slack_users = team_data.users
-          @slack_channels = team_data.channels
+          fork do
+            Lita.logger.debug("Inserting #{team_data.users.size} users")
+            UserCreator.create_users(team_data.users, robot, @robot_id)
+            Lita.logger.debug("Done inserting users.")
+
+            Lita.logger.debug("Inserting #{team_data.channels.size} channels")
+            RoomCreator.create_rooms(team_data.channels, robot)
+            Lita.logger.debug("Done inserting channels.")
+          end
```

Comment:
Seems a simple but nice approach to the problem - are those informations related? If not, what about creating another process (fork) for `teams`?

Reply:
Not sure if we will gain anything as it is being forked already, so 🤷‍♂

Reply:
I was playing around with this and it seems to work fine with two forks, but I don't really think we'd gain much here - the main thing is not to speed up the secondary process, it's just to unblock the main one. I'm going to leave as that for now and test with Spy; if it works well we can optimize this.